app/src/main/java/com/dessalines/thumbkey/utils/Utils.kt

```diff
@@ -747,6 +747,38 @@ fun autoCapitalizeCheck(ime: IMEService): Boolean {
     // Knows if its an empty field
     val empty = ime.currentInputConnection.getTextBeforeCursor(1, 0).isNullOrEmpty()
 
+    // Avoid capitalizing in certain edit boxes
+    val inputType = ime.currentInputEditorInfo.inputType and (InputType.TYPE_MASK_VARIATION)
+    if (listOf(
+            InputType.TYPE_TEXT_VARIATION_URI,
+            InputType.TYPE_TEXT_VARIATION_WEB_EMAIL_ADDRESS,
+            InputType.TYPE_TEXT_VARIATION_EMAIL_ADDRESS,
+        ).contains(inputType) || ime.currentInputEditorInfo.inputType == EditorInfo.TYPE_NULL
+    ) {
+        return false
```

Comment:
Same thing here, this should be externalized to a function, and just remove the return false.